maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultMavenPluginManager.java

```diff
@@ -216,9 +216,10 @@ private PluginDescriptor extractPluginDescriptor( Artifact pluginArtifact, Plugi
 
                     if ( pluginDescriptorEntry != null )
                     {
-                        InputStream is = pluginJar.getInputStream( pluginDescriptorEntry );
-
-                        pluginDescriptor = parsePluginDescriptor( is, plugin, pluginFile.getAbsolutePath() );
+                        try ( InputStream is = pluginJar.getInputStream( pluginDescriptorEntry ) )
```

Comment:
This is wrong.
The caller must not close the stream. It can be done on in the private method `parsePluginDescriptor()` and not here.
The method wraps the stream and so the `Reader` has to close it.

Reply:
Agree here.

Reply:
In line 232 is the bug @Tibor17 has described.

Reply:
I tend to disagree because the `Closeable#close()` method is explicitely described as being idempotent.  Else, the same applies to the call to `builder.build()` inside the `parsePluginDescriptor` which also closes the given stream.

Reply:
@gnodet You are right. Let's make it consistently correct.

Reply:
@gnodet 
It is not about the low level of `close` method.
The problem is that the second close is redundant.
And the most important thing is that the straem is wrapped by `Reader` in the method `parsePluginDescriptor()` and the method should close the `Reader`. The caller is only dispatching the stream and that's the whole thing and the caller of the method `parsePluginDescriptor()` does not need to do anything about closing the nested stream. Why to do it twice, makes no sense!

Reply:
The issue, from my POV, is that I think it is best for the method that allocates a resource to release it, and to so do "finally" with a try-with-resources statement. There are exceptions of course. What I personally do not favor is the style that I see here and there in Maven where one method allocates a stream and another method deep in the code path closes the stream; to me, this is an anti-pattern.

Reply:
@garydgregory 
It does not matter that one method is producer and another is consumer. The reason is that both are private methods and still such private methods have to by analysed. The private methods only simplify the reading abilities of the code because human brain is not able to analyse whole inlined code. Of course you can parse `pluginJar` as a method parameter into the method `parsePluginDescriptor()` but in that case we are not talking only about `try with resource` but we are talking about refactoring shich is a subject of another PR.